src/main/java/io/debezium/connector/yugabytedb/YBClientUtils.java

```diff
@@ -1,5 +1,189 @@
+    for (MasterReplicationOuterClass.GetCDCDBStreamInfoResponsePB.TableInfo tableInfo : resp.getTableInfoList()) {
+        if (Objects.equals(tableId, tableInfo.getTableId().toStringUtf8())) {
+            return true;
+        }
+    }
+
+    // This signifies that the table ID we have provided is not a part of the stream ID
+    return false;
+  }
+
+  /**
+   * Get the list of all the table UUIDs to be included for streaming
+   * @param ybClient the {@link YBClient} instance
+   * @param connectorConfig connector configuration for the connector
+   * @return a Set of the tableIDs
+   */
+  public static Set<String> fetchTableList(YBClient ybClient,
+                                           YugabyteDBConnectorConfig connectorConfig) {
+    LOGGER.info("Fetching all the tables from the source");
+    
+    Set<String> tableIds = new HashSet<>();
+      try {
+          ListTablesResponse tablesResp = ybClient.getTablesList();
+          for (MasterDdlOuterClass.ListTablesResponsePB.TableInfo tableInfo : 
+              tablesResp.getTableInfoList()) {
+              if (tableInfo.getRelationType() == MasterTypes.RelationType.INDEX_TABLE_RELATION ||
+                    tableInfo.getRelationType() == MasterTypes.RelationType.SYSTEM_TABLE_RELATION) {
+                  // Ignoring the index and system tables from getting added for streaming.
+                  continue;
+              }
+
+              // Ignore the tables without a pgschema_name, these tables are the ones created with 
+              // the older versions of YugabyteDB where the changes for CDCSDK were not present. 
+              // For more details, visit https://github.com/yugabyte/yugabyte-db/issues/11976
+              if (tableInfo.getPgschemaName() == null || tableInfo.getPgschemaName().isEmpty()) {
+                  LOGGER.warn(String.format("Ignoring the table %s.%s since it does not have" 
+                    + " a pgschema_name value (possibly because it was created using an older"
+                    + " YugabyteDB version)", tableInfo.getNamespace().getName(),
+                      tableInfo.getName()));
+                  continue;
+              }
+
+              String fqlTableName = tableInfo.getNamespace().getName() + "." 
+                                    + tableInfo.getPgschemaName() + "." 
+                                    + tableInfo.getName();
+              TableId tableId = YugabyteDBSchema.parseWithSchema(fqlTableName, 
+                                                                 tableInfo.getPgschemaName());
+
+              // Retrieve the list of tables in the stream ID,
+              GetDBStreamInfoResponse dbStreamInfoResponse = ybClient.getDBStreamInfo(
+                                                               connectorConfig.streamId());
+
+              if (connectorConfig.getTableFilters().dataCollectionFilter().isIncluded(tableId)
```

Comment:
Yes, for now it will fail. Note that this was the existing behavior, we have just refactored the function to a different utility class.

This behavior will be changed/fixed in the PR which will have the add table support.